packages/fuels-core/src/code_gen/custom_types_gen.rs

```diff
@@ -69,7 +69,7 @@ pub fn expand_custom_struct(prop: &Property) -> Result<TokenStream, Error> {
                 fields.push(quote! {pub #field_name: #enum_name});
                 args.push(quote! {#field_name: #enum_name::new_from_tokens(&tokens[#idx..])});
                 struct_fields_tokens.push(quote! { tokens.push(self.#field_name.into_token()) });
-                param_types.push(quote! { types.push(ParamType::Enum(#enum_name::param_types())) });
+                param_types.push(quote! { types.push(ParamType::Enum(EnumVariants::new(#enum_name::param_types()).unwrap())) });
```

Comment:
Using '?' would require changing the signature of:
`fn param_types() -> Vec<ParamType>` 
to
`fn param_types() -> Result<ParamType, Error>`

since these param_types are expanded into the `param_types` method on the custom type.

But it is safe to call unwrap() here because: 
We're going through all the components that make up some custom type. For every component, we parse it into a `ParamType` and proceed to do various things depending on which ParamType it is.

Here we've matched against `ParamType::Enum(EnumVariants)` so we know for a fact that the component (enum) in question must have variants since `EnumVariants` could not have been constructed empty.

Since we know that we successfully created EnumVariants during compile time for this type, we can safely do so later on in runtime when somebody calls `param_types()` on the generated type.

Reply:
Perfect justification! can you add this reasoning somewhere in the docstrings? I saw you do that reasoning with Ahmed above and I think it's worth writing it down in code.